components/device-mgt/org.wso2.carbon.device.mgt.api/src/main/java/org/wso2/carbon/device/mgt/jaxrs/service/impl/DeviceOrganizationServiceImpl.java

```diff
@@ -0,0 +1,250 @@
+        if (deviceOrganizationMetadataHolder == null) {
+            String errorMessage = "The payload of device metadata is empty.";
+            return Response.status(Response.Status.BAD_REQUEST).entity(errorMessage).build();
+        }
+        DeviceOrganizationProviderService dops = new DeviceOrganizationProviderServiceImpl();
+        String tempId = deviceOrganizationMetadataHolder.getDeviceId();
+        String tempName = deviceOrganizationMetadataHolder.getDeviceName();
+        String tempParent = deviceOrganizationMetadataHolder.getParent();
+        int tempPingMins = deviceOrganizationMetadataHolder.getPingMins();
+        int tempState = deviceOrganizationMetadataHolder.getState();
+        int tempIsGateway = deviceOrganizationMetadataHolder.getIsGateway();
+        boolean result;
+        try {
+            result  = dops.addDeviceOrganization(tempId,tempName,tempParent,tempPingMins,tempState,tempIsGateway);
+            return Response.status(Response.Status.OK).entity(result).build();
+        } catch (DeviceOrganizationException e) {
+            String msg = "Error occurred while enrolling the device with ID:" + tempId + " to Device Organization ";
+            log.error(msg, e);
+            return Response.status(Response.Status.INTERNAL_SERVER_ERROR).entity(msg).build();
+        } catch (InvalidConfigurationException e) {
+            log.error("failed to add operation", e);
+            return Response.status(Response.Status.INTERNAL_SERVER_ERROR).build();
+        }
+    }
+
+    @GET
+    @Path("/{deviceId}/state")
+    @Override
+    public Response getDeviceOrganizationStateById(@PathParam("deviceId") String deviceId) {
+        if (deviceId.length() <= 0) {
+            String errorMessage = "The parameter of the device organization ID is empty.";
+            return Response.status(Response.Status.BAD_REQUEST).entity(errorMessage).build();
+        }
+        int result;
+        DeviceOrganizationProviderService dops = new DeviceOrganizationProviderServiceImpl();
+        try {
+            result = dops.getDeviceOrganizationStateById(deviceId);
+            if (result == -1) {
+                String msg = "Device with ID: " + deviceId + " not found.";
+                log.error(msg);
+                return Response.status(Response.Status.NOT_FOUND).entity(msg).build();
+            }
+            return Response.status(Response.Status.OK).entity(result).build();
+        } catch (DeviceOrganizationException e) {
+            String msg = "Error occurred while retrieving state of device with ID:" + deviceId + " to Device Organization ";
+            log.error(msg, e);
+            return Response.status(Response.Status.INTERNAL_SERVER_ERROR).entity(msg).build();
+        } catch (InvalidConfigurationException e) {
+            log.error("failed to add operation", e);
+            return Response.status(Response.Status.INTERNAL_SERVER_ERROR).build();
```

Comment:
Please add message entity here by describing the error.

Reply:
Fixed 807397142da40a2c9231a718c14619e1442b02e2